01-overview-of-ml-systems.md

```diff
@@ -0,0 +1,110 @@
+3. **The task happens at scale:** deploying ML models to production is hard. This work is more justifiable when the model will be used to make lots of predictions. Also, if the task happens at scale, there will be lots of data to train from.
+4. **The patterns are constantly changing:** ML shines against traditional software rules when patterns change frequently, causing hardcoded rules get outdated too fast.  Changes in spam trends and culture trends are examples of changing patterns.
+
+
+
+## Typical ML use cases
+ML is being used both for **consumer** facing applications and **enterprise** internal applications. Consumer ML applications is growing; however, as of 2022, the vast majority of applications is still in the enterprise space.
+
+- In **consumer** applications latency tends to be more important than accuracy. Increases in latency may cause the user to bounce from our app. Consumers also tend to be more forgiving with bad predictions in non-critical applications like recommending a movie or bad tagging of a photo.
+- In **enterprise** applications, accuracy tends to be more important than latency. Small increases in accuracy can save millions of dollars per year.  The image below shows the breakdown of ML enterprise applications. These are some examples:
+  - Use ML to reduce cost of customer acquisition (CAC): show better targeted ads, show discounts at the right moment. On  2019, the average CAC for a user that makes an in-app purchase is US $86 61. High CAC is a startup killer.
+  - Churn prediction:  Acquiring a new customer is 5 to 25 times more expensive than retaining one.
+  - Brand monitoring: Monitor the sentiment of the public in social media towards your brand. 
+
+![state-of-enterprise-ml](01-overview-of-ml-systems.assets/state-of-enterprise-ml.png)
+
+
+
+# ML in research vs production
+Research ML and production ML are different. It is important to be aware of the differences to avoid the pitfalls that arise from trying to use a research mentality to production models.
+
+|                        | Research                                                     | Production                                             |
+| ---------------------- | ------------------------------------------------------------ | ------------------------------------------------------ |
+| Requirements           | State-of-the-art model performance with benchmark datasets (1) | Different stakeholders have different requirements (2) |
+| Computational priority | Fast training, high query throughput (3)                     | Fast inference with low latency (4) (5) (6)            |
+| Data                   | Often Static                                                 | Constantly shifting                                    |
+| Fairness               | Often not a focus                                            | Must be considered (7)                                 |
```

Comment:
Computational Priority, Interpretability, and Fairness do not show much difference between enterprise and research because they do not require private data and the researchers are doing it for the enterprise so they are aligned on the goals.

Data is a problem since it's custom to the enterprise, but I am almost sure researchers have enough data to at least learn about the approaches to either how to train with private data or transfer learning.